presentation/src/main/java/com/example/grosharies/presentation/groceryList/GroceryListViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.example.grosharies.presentation.groceryList
+
+import android.app.Application
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.*
+import com.example.grosharies.data.GroshariesRoomDatabase
+import com.example.grosharies.data.groceryList.GroceryList
+import com.example.grosharies.data.groceryList.GroceryListRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class GroceryListViewModel(application: Application) : AndroidViewModel(application) {
+    private val repository: GroceryListRepository
+
+    private val mutableGroceryList: MutableLiveData<String> = MutableLiveData()
+    val groceryLists: LiveData<List<GroceryList>>
+    val groceryList: MutableState<GroceryList?> = mutableStateOf(null)
+
+    /*
+     * initialize the repository and the groceryLists
+     */
+    init {
+        val groceryListDao = GroshariesRoomDatabase.getDatabase(application)!!.groceryListDao()
+        repository = GroceryListRepository(groceryListDao)
+
+        groceryLists = Transformations.switchMap(mutableGroceryList) { param ->
+            if (param == "null") {
+                repository.getGroceryListsWithoutGroup()
+            } else {
+                repository.getGroceryLists(param)
+            }
+        }
+    }
+
+    fun getListItemsByGroup(groupId: String) {
+        mutableGroceryList.value = groupId
+    }
+
+    fun getListItemsWithoutGroup() {
+        mutableGroceryList.value = "null"
+    }
+
+    fun getGroceryListById(id: Int) {
```

Comment:
But for fetching the groceryList (also a Room action in you rrepository) you're not using coroutines, so blocking the main thread!! Ouch !